internal/runners/initialize/init.go

```diff
@@ -287,7 +287,10 @@ func deriveVersion(getKnownVersions knownVersionsFunc, lang language.Language, v
 		return "", errs.Wrap(err, "Unable to get known versions for language %s", lang.Requirement())
 	}
 
-	prefix := strings.Replace(strings.Replace(version, ".x", "", 1), ".X", "", 1) // strip any wildcard
+	prefix, _, _ := strings.Cut(version, ",") // only keep first part of e.g. ">=3.10,<3.11"
```

Comment:
I think we should drop all of this language verification logic we have here. With complex version numbers this gets way too involved for the client. We should just defer this to the buildplanner.

So if the user specified a version that can't be solved then that'll resolve in an error while sourcing the runtime. We should make sure that in this scenario the buildplanner gives an actionable error to the user though, and if not we need to file an enhancement with extensibility.

Reply:
I don't think we can drop this basic logic because we need to figure out if a given version number is a partial one or not. For example: `state init org/project --language python@3.10` without any version checking would end up installing python 3.10.0 (not 3.10.12 or whatever the latest or recommended point release is; the solver appears to just tack on a ".0" to the end of what we gave it). The logic currently in place would detect that "3.10" is not a full version number and would append ".x" to the end, resulting in "3.10.x", which gives the latest point release.

Reply:
That is a problem for the API to address though, not us. And keep in mind that most users won't specify a version, this is already an advanced case. I don't want to have special solving logic around to address API limitations as it'll just lead to the API's never addressing those limitations and us having special workaround logic that needs to be maintained.